sherpa/astro/datastack/__init__.py

```diff
@@ -1086,40 +312,56 @@ def wrap(*args, **kwargs):
         return wrapfunc(*args, **kwargs)
 
     wrap.__name__ = func.__name__
-    wrap.__doc__ = func.__doc__
+    if not hasattr(DataStack, func.__name__):
```

Comment:
Can the logic be swapped here, as to me it reads  better - i.e.

    if hasattr(...):
        doc = getattr(...)
    else:
        doc = func.__doc__